lib/WUI/nhttp/gcode_upload.h

```diff
@@ -46,14 +46,22 @@ class GcodeUpload final : private UploadHooks {
     virtual Result finish(const char *final_filename, bool start_print) override;
     virtual Result check_filename(const char *filename) const override;
 
+    // filepath in case of PUT (we know it from the start)
+    char filepath [FILE_PATH_BUFFER_LEN];
+    bool print_after_upload;
+
+    handler::Step stepPOST(std::string_view input);
+    handler::Step stepPUT(std::string_view input);
+
     GcodeUpload(UploadState uploader, bool json_errors, size_t length, size_t upload_idx, unique_file_ptr file, UploadedNotify *uploaded);
+    GcodeUpload(bool json_errors, size_t length, size_t upload_idx, unique_file_ptr file, UploadedNotify *uploaded, const char* filepath, bool print_after_upload);
 
 public:
     bool want_read() const { return size_rest > 0; }
     bool want_write() const { return false; }
     handler::Step step(std::string_view input, bool terminated_by_client, uint8_t *output, size_t output_size);
     using UploadResult = std::variant<handler::StatusPage, GcodeUpload>;
-    static UploadResult start(const handler::RequestParser &parser, UploadedNotify *uploaded, bool json_errors);
+    static UploadResult start(const handler::RequestParser &parser, UploadedNotify *uploaded, bool json_errors, const char* filename = nullptr, bool print_after_upload = false);
```

Comment:
Do I understand correctly that in case of `POST`, you (implicitly) set `print_after_upload` to `false`, but still may print after upload? That sounds like something a bit confusing. The situation there isn't „We don't want to print after upload“, but „We don't know if we want to print after upload or not“.

Furthermore, considering we want to add more such flags (eg. `overwrite`), I'm also not entirely sure adding a parameter is the right way or if we should just pull it out of the parser that's already passed in… similarly to pulling out the boundary.